src/Coin.cpp

```diff
@@ -0,0 +1,264 @@
+
+    case TWCoinTypeRipple:
+        return Ripple::Address::isValid(string);
+
+    case TWCoinTypeDash:
+        return Bitcoin::Address::isValid(string, {TWP2PKHPrefixDash, TWP2SHPrefixDash});
+
+    case TWCoinTypeTezos:
+        return Tezos::Address::isValid(string);
+
+    case TWCoinTypeTron:
+        return Bitcoin::Address::isValid(string, {0x41});
+
+    case TWCoinTypeZcoin:
+        return Bitcoin::Address::isValid(string, {TWP2PKHPrefixZcoin, TWP2SHPrefixZcoin});
+
+    case TWCoinTypeZcash:
+        return Zcash::TAddress::isValid(string, {TWP2PKHPrefixZcashT, TWP2SHPrefixZcashT});
+
+    case TWCoinTypeEOS:
+        return false;
+    }
+}
+
+TWPurpose TW::purpose(TWCoinType coin) {
+    switch(coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeNimiq:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return TWPurposeBIP44;
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeLitecoin:
+        return TWPurposeBIP84;
+    }
+}
+
+TWCurve TW::curve(TWCoinType coin) {
+    switch(coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeLitecoin:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return TWCurveSECP256k1;
+    case TWCoinTypeNimiq:
+        return TWCurveEd25519;
+    }
+}
+
+DerivationPath TW::derivationPath(TWCoinType coin) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeLitecoin:
+    case TWCoinTypeNimiq:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return DerivationPath(purpose(coin), coin, 0, 0, 0);
```

Comment:
This is default case, I think we may need another methods:
1. accept `account`, `change`, `index`
2. only 3 levels for Ed25519

Reply:
We do support that, just none implemented yet.